WordPress/Classes/ViewRelated/Themes/ThemeBrowserCell.swift

```diff
@@ -193,10 +193,20 @@ open class ThemeBrowserCell: UICollectionViewCell {
         activityView.stopAnimating()
     }
 
-    fileprivate func refreshScreenshotImage(_ imageUrl: String) {
+    private func getImageUrlForWidth(imageUrl: String) -> String {
         // Themes not hosted on WP.com have an incorrect screenshotUrl and do not correctly support the w param
-        let imageUrlForWidth = imageUrl.hasPrefix("http") ? imageUrl + "?w=\(presenter!.screenshotWidth)" :
-            String(format: "http:%@", imageUrl)
+        if false == imageUrl.hasPrefix("http") {
+            return String(format: "http:%@", imageUrl)
+        }
+
+        if false == imageUrl.contains("?") {
+            return imageUrl + "?w=\(presenter!.screenshotWidth)"
+        }
+        return imageUrl + "&w=\(presenter!.screenshotWidth)"
```

Comment:
For me this method is a bit hard to read and understand what's happening. I wonder if we can clean it up with a guard statement: 
```suggestion
        guard imageUrl.hasPrefix("http") else {
            return String(format: "http:%@", imageUrl)
        }

        if !imageUrl.contains("?") {
            return imageUrl + "?w=\(presenter!.screenshotWidth)"
        }
        return imageUrl + "&w=\(presenter!.screenshotWidth)"
```


It seems like the goal here is to produce a URL where we append a new query parameter. I wonder if we'll ever want to include a set height or include the devices resolution as well to ensure consistent images. If we wanted this to return back the full URL we could enhance this to be:

```
var components = URLComponents(string: imageUrl)
components?.queryItems?.append(URLQueryItem(name: "w", value: "\(presenter!.screenshotWidth)"))
components?.url
```

As a follow-up note, I do have some issues with the need to force unwrap the presenter with `presenter!`. I realize this was in place beforehand and that a crash here is an edge case but we can probably conditionally unwrap this and not add the parameter if the size isn't available.

Reply:
Thanks Chip
I update the method to use the URLComponents.